cpp/src/io/text/bgzip_data_chunk_source.cu

```diff
@@ -0,0 +1,456 @@
+    rmm::device_uvector<compression_result> d_decompression_results;
+    std::size_t compressed_size_with_headers{};
+    std::size_t max_decompressed_size{};
+    // this is usually equal to decompressed_size()
+    // unless we are in the last chunk, where it's limited by _local_end
+    std::size_t available_decompressed_size{};
+    std::size_t read_pos{};
+    bool decompressed{};
+
+    compressed_blocks(rmm::cuda_stream_view init_stream)
+      : d_compressed_blocks(0, init_stream),
+        d_decompressed_blocks(0, init_stream),
+        d_compressed_offsets(0, init_stream),
+        d_decompressed_offsets(0, init_stream),
+        d_compressed_spans(0, init_stream),
+        d_decompressed_spans(0, init_stream),
+        d_decompression_results(0, init_stream)
+    {
+      CUDF_CUDA_TRY(cudaEventCreate(&event));
+      h_compressed_blocks.reserve(default_buffer_alloc);
+      h_compressed_offsets.reserve(default_offset_alloc);
+      h_compressed_offsets.push_back(0);
+      h_decompressed_offsets.reserve(default_offset_alloc);
+      h_decompressed_offsets.push_back(0);
+    }
+
+    void decompress(rmm::cuda_stream_view stream)
+    {
+      if (decompressed) { return; }
+      copy_to_device(h_compressed_blocks, d_compressed_blocks, stream);
+      copy_to_device(h_compressed_offsets, d_compressed_offsets, stream);
+      copy_to_device(h_decompressed_offsets, d_decompressed_offsets, stream);
+      d_decompressed_blocks.resize(decompressed_size(), stream);
+      d_compressed_spans.resize(num_blocks(), stream);
+      d_decompressed_spans.resize(num_blocks(), stream);
+      d_decompression_results.resize(num_blocks(), stream);
+
+      auto offset_it = thrust::make_zip_iterator(d_compressed_offsets.begin(),
+                                                 d_compressed_offsets.begin() + 1,
+                                                 d_decompressed_offsets.begin(),
+                                                 d_decompressed_offsets.begin() + 1);
+      auto span_it =
+        thrust::make_zip_iterator(d_compressed_spans.begin(), d_decompressed_spans.begin());
+      thrust::transform(
+        rmm::exec_policy_nosync(stream),
+        offset_it,
+        offset_it + num_blocks(),
+        span_it,
+        bgzip_nvcomp_transform_functor{reinterpret_cast<uint8_t const*>(d_compressed_blocks.data()),
+                                       reinterpret_cast<uint8_t*>(d_decompressed_blocks.begin())});
+      if (decompressed_size() > 0) {
+        cudf::io::nvcomp::batched_decompress(cudf::io::nvcomp::compression_type::DEFLATE,
+                                             d_compressed_spans,
+                                             d_decompressed_spans,
+                                             d_decompression_results,
+                                             max_decompressed_size,
+                                             decompressed_size(),
+                                             stream);
```

Comment:
You should follow the pattern in other components (use nvCOMP only if it's enabled). Example:
```
if (nvcomp_integration::is_all_enabled()) {
          nvcomp::batched_decompress(nvcomp::compression_type::DEFLATE,
                                     inflate_in_view,
                                     inflate_out_view,
                                     inflate_res,
                                     max_uncomp_block_size,
                                     total_decomp_size,
                                     stream);
        } else {
          gpuinflate(
            inflate_in_view, inflate_out_view, inflate_res, gzip_header_included::NO, stream);
        }
```
This should fix the tests. nvCOMP DEFLATE is not enabled by default, so other formats are working fine with nvCOMP 2.4.

Reply:
that fixed it